### ISSUE TITLE
Support e2e tests / example YAMLS for behaviour behind feature flags

### ISSUE THREAD

**Issue by sbwsg** [tektoncd/pipeline#2574]
# Expected Behavior

I should be able to write e2e tests or YAML examples that require feature flags be set a specific way.

# Actual Behavior

Currently doing so would potentially break other tests running in the same cluster since the feature flags are cluster-scoped.

# Additional Info

Couple ideas that have been floated to fix this so far:

1. What if Tekton Pipelines could be set to only watch individual namespaces?
    - this would allow a test to set feature flags that are isolated solely to its own pipelines controller.
    - added benefit: multiple tekton pipelines deployments could live in a single cluster
2. What if feature flags could be set on a per-namespace basis?
    - this would allow a test to set feature flags isolated solely to their own testing namespace (i.e. "arendelle-XXXX")

**Comment by sbwsg** [comment]
Reopening as I'd still like Tekton to be able to test code gated behind feature flags.

/reopen
/remove-lifecycle stale
/remove-lifecycle rotten

**Comment by bobcatfish** [comment]
I'd like to see an approach here that avoids having to run the end to end tests for every combination of feature flags; e.g. specific tests that are targeted at each flag. @vdemeester was mentioning using annotations to turn feature flags on and off in some cases, that might make it easier to make tests aimed at each flag.

if we DO decide we need to run all the tests for each flag (and combo) maybe we can do it nightly or similar, i.e. not on every PR

**Comment by sbwsg** [comment]
Yeah, agreed.  I'm personally much less concerned about the interaction of flag combinations than I am about being able to e2e test code behind flags _at all_.  I like the annotations idea!

**Comment by vdemeester** [comment]
/reopen
/lifecycle frozen
We should tackle this still
